### ISSUE TITLE
Add conversions from BitSlice into integers

### ISSUE THREAD

**Issue by danieldulaney** [myrrlyn/bitvec#31]
Hi! I was using this crate, and I found a need to convert `BitSlice`s back into integers of various lengths. This is different from the already-existing `as_slice` -- that gives direct access to the underlying storage, which might include bits on either side of the `BitSlice` if the `BitSlice` starts or ends mid-byte.

What I would like is the ability to convert from a `BitSlice` to a regular Rust integer, with `bitvec` performing whatever shifts need to happen behind the scenes.

I could put together a patch that would implement `TryInto<u8>`, `TryInto<u16>`, etc. for `BitSlice` types. Is that something that would be welcome?

**Comment by myrrlyn** [comment]
I am working towards 0.16 away from master, largely because I have a lot of technical debt to clean up; on this branch I have the beginnings of a value load/store behavior: [`src/fields.rs`](https://github.com/myrrlyn/bitvec/blob/feature/api-refactor/src/fields.rs).

Currently, it is restricted to the underlying type; I am still working on the design of how`BitSlice` can support trait-based load/store behavior for arbitrary types.

If I cannot resolve a design I like in the near term, I will publish 0.16 with the current storage-type-only behavior and leave the rest for future work.

I do not provide stability guarantees about tracking the git repository in your crate manifest, primarily because I am just bad enough at git to have mistakes in the tree and just good enough at it to be able to erase them.

Thank you for filing this issue; it's nice to have external confirmation that my experiments are going in useful directions.

I do not plan to implement `TryInto` at this time, because `[T]` is not `TryInto<T>`, and I would like `BitSlice` to not implement stdlib traits that slices do not implement; API mirroring means matching what the source has and also what it doesn't.

Let me know if the current implementation is sufficient; if it is, I will speed up the 0.16 release.

**Comment by myrrlyn** [comment]
Bitfield access is now implemented as generic over all `BitStore` types, decoupled from the underlying store of a slice. Any unsigned integer can be loaded from or stored into a slice of any backing type, as long as the slice is not wider than the transfer type.

My reliance on `BitStore` means that floating-point numbers cannot be transferred as-is, and `u64` is not supported on 32-bit systems. I will address this in the future.

Once I have finished the test suite, I will push this to master and close the issue. I will release 0.16 with this feature to crates.io before the end of the year.
